include/sph/kernels.hpp

```diff
@@ -51,16 +56,39 @@ CUDA_DEVICE_FUN inline T artificial_viscosity(T ro_i, T ro_j, T h_i, T h_j, T c_
     T c_ij  = (c_i + c_j) * T(0.5);
     T h_ij  = (h_i + h_j) * T(0.5);
 
-    // calculate viscosity_ij according to Monaghan & Gringold 1983
     T viscosity_ij = 0.0;
     if (rv < 0.0)
     {
-        // calculate muij
         T mu_ij = (h_ij * rv) / (r_square + epsilon * h_ij * h_ij);
         viscosity_ij  = (-alpha * c_ij * mu_ij + beta * mu_ij * mu_ij) / ro_ij;
     }
 
     return viscosity_ij;
 }
 
+/*! @brief calculate the artificial viscosity between a pair of two particles
+ *
+ * @tparam T    float or double
+ * @param c_i   speed of sound particle i
+ * @param c_j   speed of sound particle j
+ * @param w_ij  relative velocity (v_i - v_j), projected onto the connecting axis (r_i - r_j)
+ * @return      the viscosity
+ */
+template<typename T>
+CUDA_DEVICE_FUN inline T artificial_viscosity(T c_i, T c_j, T w_ij)
+{
+    // these are const now, but will be different for each particle when using viscosity switching
```

Comment:
Maybe state here that only alfa will be variable. Beta will always be 2.

Reply:
ok, will do.